Doc/library/socket.rst

```diff
@@ -631,6 +647,18 @@ Constants
 
    .. versionadded:: 3.12
 
+.. _socket-ethernet-types:
+
+.. data:: ETHERTYPE_ARP
+          ETHERTYPE_IP
+          ETHERTYPE_IPV6
+          ETHERTYPE_VLAN
+
+   IEEE 802.3 protocol numbers [1]_.
```

Comment:
Instead of making the reader clicking through to a footnote that just contains a bare link they would then need to click, why not just make this itself a proper link?

```suggestion
   `IEEE 802.3 protocol number
   <https://www.iana.org/assignments/ieee-802-numbers/ieee-802-numbers.txt>`_.
   constants.
```

Furthermore, assuming I understand this correctly (if I don't, just use the previous suggestion), wouldn't it be useful to add a cross-reference on their use with the module?

```suggestion
   `IEEE 802.3 protocol number
   <https://www.iana.org/assignments/ieee-802-numbers/ieee-802-numbers.txt>`_.
   constants.
   Can be passed to *proto* in the :class:`~socket.socket` constructor.
```